### PR TITLE
Add Finnish PO file (Engine)

### LINKED ISSUE #879: Add Finnish translation
Add an `fi.po` for all test messages.

### PR DESCRIPTION
**Pull request by matsduf**
This PO file has been sent by email by translator. It needs some corrections before it can be merged. This resolves #879.

### CODE REVIEW

File: share/fi.po [hunk 1]
```diff
@@ -0,0 +1,2068 @@
...
+"Delegation lists no nameserver that resolves to an IPv6 address. If any were "
+"present, the minimum allowed would be {minimum}."
+msgstr ""
+"Delegointi ei listaa yhtään nimipalvelinta, joka vastaisi IPv6-osoitetta. "
+"Jos niitä olisi olemassa, alin sallittu määrä olisi {minimum}."
+
+#. DELEGATION:NS_IS_CNAME
+#, fuzzy, perl-brace-format
+msgid "Nameserver {nsname} RR points to CNAME."
+msgstr "Nimipalvelimen {nsname} RR osoittaa tietueeseen CNAME."
```
Comment (matsduf) [inline share/fi.po:1277]: This translation is marked as "fuzzy". It must be resolved by the translator.

File: share/fi.po [hunk 2]
```diff
@@ -0,0 +1,2068 @@
...
+"postipalvelinta, jolla on IP-osoite."
+
+#. SYNTAX:RNAME_MAIL_DOMAIN_LOCALHOST
+#, fuzzy, perl-brace-format
+msgid ""
+"The SOA RNAME mail domain ({domain}) resolved to a mail server with "
+"localhost ({localhost}) IP address."
+msgstr ""
+"SOA RNAME postin verkkotunnukselle ({domain}) ei löydy vastaavaa "
+"postipalvelinta, jolla on IP-osoite."
```
Comment (matsduf) [inline share/fi.po:1724]: This translation is marked as "fuzzy".

Also, `../util/check-msg-args fi.po` says "Only in msgid: {localhost}".

Both must be resolved by the translator.

File: share/fi.po [hunk 3]
```diff
@@ -0,0 +1,2068 @@
...
+
+#. SYNTAX:RNAME_MAIL_ILLEGAL_CNAME
+#, fuzzy, perl-brace-format
+msgid ""
+"The SOA RNAME mail domain ({domain}) refers to an address which is an alias "
+"(CNAME)."
+msgstr "SOA 'mname' -arvo ({mname}) viittaa NS:ään, joka on alias (CNAME)."
```
Comment (matsduf) [inline share/fi.po:1731]: This translation is marked as "fuzzy". 

Also, `../util/check-msg-args fi.po says` "Only in msgid: {domain}" and "Only in msgstr: {mname}".

Both must be resolved by the translator.

**Comment by matsduf** [comment]
@samisalmensuo, can you look at this?

File: share/fi.po (hunk 2 above)
Comment (samisalmensuo) [reply share/fi.po:1724]: Correct:

msgid ""
"The SOA RNAME mail domain ({domain}) resolved to a mail server with "
"localhost ({localhost}) IP address."
msgstr ""
"SOA RNAME postin verkkotunnukselle ({domain}) ei löydy vastaavaa "
"postipalvelinta, jolla on IP-osoite ({localhost})."

File: share/fi.po (hunk 3 above)
Comment (samisalmensuo) [reply share/fi.po:1731]: Correct:

msgid ""
"The SOA RNAME mail domain ({domain}) refers to an address which is an alias "
"(CNAME)."
msgstr ""
"SOA RNAME sähköpostiosoite ({domain}) viittaa osoitteeseen joka on alias " 
"(CNAME)."

**Review by samisalmensuo — COMMENTED** [review]
hope i understood the problems correctly. :)

**Comment by matsduf** [comment]
Uppdated with corrections provided by @samisalmensuo. Fuzzy marks then removed.

@samisalmensuo, the fuzzy marks were set by the gettext program when it provided msgstr strings by guessing.

**Comment by matsduf** [comment]
@pnax and @mattias-p, can you approve?

**Review by mattias-p — APPROVED** [review]

**Comment by matsduf** [comment]
Rebased on develop branch with corrected unit test data. No change of the PR as such.

**Review by mattias-p — APPROVED** [review]

**Comment by matsduf** [comment]
To test this PR, follow the [installation instructions] with the following modification:

1. Fetch [Zonemaster-Engine-v4.1.0-PR-880-a2553ae-2021-03-17.tar.gz]
2. Replace `cpanm Zonemaster::Engine` with
```
cpanm Zonemaster-Engine-v4.1.0-PR-880-a2553ae-2021-03-17.tar.gz
```

[Zonemaster-Engine-v4.1.0-PR-880-a2553ae-2021-03-17.tar.gz]: https://github.com/zonemaster/zonemaster-engine/files/6157856/Zonemaster-Engine-v4.1.0-PR-880-a2553ae-2021-03-17.tar.gz




[installation instructions]: https://github.com/zonemaster/zonemaster-engine/blob/master/docs/Installation.md

**Pull request merged**